src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py

```diff
@@ -0,0 +1,1658 @@
+                )  # .split(self.all_head_size, dim=-1)
+
+        score = 0
+        # content->position
+        if "c2p" in self.pos_att_type:
+            scale = tf.math.sqrt(tf.cast(shape_list(pos_key_layer)[-1] * scale_factor, tf.float32))
+            c2p_att = tf.matmul(query_layer, tf.transpose(pos_key_layer, [0, 2, 1]))
+            c2p_pos = tf.clip_by_value(relative_pos + att_span, 0, att_span * 2 - 1)
+            c2p_att = take_along_axis(
+                c2p_att,
+                tf.broadcast_to(
+                    tf.squeeze(c2p_pos, 0),
+                    [shape_list(query_layer)[0], shape_list(query_layer)[1], shape_list(relative_pos)[-1]],
+                ),
+                -1,
+            )
+            score += c2p_att / scale
+
+        # position->content
+        if "p2c" in self.pos_att_type or "p2p" in self.pos_att_type:
+            scale = tf.math.sqrt(tf.cast(shape_list(pos_query_layer)[-1] * scale_factor, tf.float32))
+            if shape_list(key_layer)[-2] != shape_list(query_layer)[-2]:
+                r_pos = build_relative_position(
+                    shape_list(key_layer)[-2],
+                    shape_list(key_layer)[-2],
+                    bucket_size=self.position_buckets,
+                    max_position=self.max_relative_positions,
+                )
+                r_pos = tf.expand_dims(r_pos, 0)
+            else:
+                r_pos = relative_pos
+
+            p2c_pos = tf.clip_by_value(-r_pos + att_span, 0, att_span * 2 - 1)
+
+        if "p2c" in self.pos_att_type:
+            p2c_att = tf.matmul(key_layer, tf.transpose(pos_query_layer, [0, 2, 1]))
+            p2c_att = tf.transpose(
+                take_along_axis(
+                    p2c_att,
+                    tf.broadcast_to(
+                        tf.squeeze(p2c_pos, 0),
+                        [shape_list(query_layer)[0], shape_list(key_layer)[-2], shape_list(key_layer)[-2]],
+                    ),
+                    -1,
+                ),
+                [0, 2, 1],
+            )
+            score += p2c_att / scale
+
+        # position->position
+        if "p2p" in self.pos_att_type:
+            pos_query = pos_query_layer[:, :, att_span:, :]
+            p2p_att = tf.matmul(pos_query, tf.transpose(pos_key_layer, [0, 2, 1]))
+            p2p_att = tf.broadcast_to(shape_list(query_layer)[:2] + shape_list(p2p_att)[2:])
+            p2p_att = take_along_axis(
+                p2p_att,
+                tf.broadcast_to(
+                    c2p_pos,
+                    [
+                        shape_list(query_layer)[0],
+                        shape_list(query_layer)[1],
+                        shape_list(query_layer)[2],
+                        shape_list(relative_pos)[-1],
+                    ],
+                ),
+                -1,
+            )
+            score += p2p_att
+
+        return score
+
+
+class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```

Comment:
This looks identical to the `TFDebertaEmbeddings` that you have implemented in the TF DeBERTa implementation. Would you mind adding a comment mentioning it's a direct copy? We have CI in place to ensure that the two copies do not diverge
```suggestion
# Copied from transformers.models.deberta.modeling_deberta.TFDebertaEmbeddings with xxx->yyy
class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```
You can replace the `xxx` and `yyy` with the appropriate changes, for example here I believe the `StableDropout` changes so you would put
```suggestion
# Copied from transformers.models.deberta.modeling_deberta.TFDebertaEmbeddings with TFDebertaStableDropout->TFDebertaV2StableDropout
class TFDebertaV2Embeddings(tf.keras.layers.Layer):
```

Reply:
done